File: src/ruby/NinjaSurvivalRate/src/impact_calculator.rb
class ImpactCalculator
  def impact (ninja, opponent, weapon, target)
    if ninja.nil? 
      raise "Ninja is not defined"
    end
    if ninja.belt? != "third" 
      return "split" 
    end
    if opponent=="Chuck Norris"
      return "split" 
    end
    "not harmed"
  end
end

File: src/dotnet/NinjaSurvivalRate/src/ImpactCalculator.cs
using System;

namespace NinjaSurvivalRate
{
    public class ImpactCalculator
    {
        public string CalculateImpact(Ninja ninja, string opponent)
        {
            if (ninja == null)
                throw new Exception("Ninja is not defined");
            if (ninja.Belt != "third")
                return "split";
            if (opponent == "chuck Norris")
                return "split";
            return "not harmed";
        }
    }
}